plugin/trino-iceberg/src/test/java/io/trino/plugin/iceberg/BaseIcebergMaterializedViewTest.java

```diff
@@ -594,13 +593,134 @@ public void testStorageSchemaProperty()
+        assertThat(query("SELECT " + exampleValue))
+                .matches("SELECT CAST(%s AS %S)".formatted(exampleValue, dataType));
+
+        assertUpdate("CREATE MATERIALIZED VIEW test_truncate_partitioning WITH (partitioning=ARRAY['truncate(col, 4)']) AS SELECT * FROM (VALUES CAST(NULL AS %s), %s) t(col)"
+                .formatted(dataType, exampleValue));
+        try {
+            SchemaTableName storageTable = getStorageTable("test_truncate_partitioning");
+            assertThat((String) computeScalar("SHOW CREATE TABLE " + storageTable))
+                    .contains("partitioning = ARRAY['truncate(col, 4)']");
+
+            assertThat(query("SELECT * FROM test_truncate_partitioning WHERE col = " + exampleValue))
+                    .matches("SELECT " + exampleValue);
+        }
+        finally {
+            assertUpdate("DROP MATERIALIZED VIEW test_truncate_partitioning");
+        }
+    }
+
+    @DataProvider
+    public Object[][] testTruncatePartitioningDataProvider()
+    {
+        // Iceberg supports truncate partitioning on int, long, decimal, string
+        return new Object[][] {
+                {"integer", "20050909"},
+                {"bigint", "200509091331001234"},
+                {"decimal(8,5)", "DECIMAL '876.54321'"},
+                {"decimal(28,21)", "DECIMAL '1234567.890123456789012345678'"},
+                {"varchar", "VARCHAR 'Greetings from Warsaw!'"},
+        };
+    }
+
+    @Test(dataProvider = "testTemporalPartitioningDataProvider")
+    public void testTemporalPartitioning(String partitioning, String dataType, String exampleValue)
+    {
+        // validate the example value type
+        assertThat(query("SELECT " + exampleValue))
+                .matches("SELECT CAST(%s AS %S)".formatted(exampleValue, dataType));
+
+        assertUpdate("CREATE MATERIALIZED VIEW test_temporal_partitioning WITH (partitioning=ARRAY['%s(col)']) AS SELECT * FROM (VALUES CAST(NULL AS %s), %s) t(col)"
+                .formatted(partitioning, dataType, exampleValue));
+        try {
+            SchemaTableName storageTable = getStorageTable("test_temporal_partitioning");
+            assertThat((String) computeScalar("SHOW CREATE TABLE " + storageTable))
+                    .contains("partitioning = ARRAY['%s(col)']".formatted(partitioning));
+
+            assertThat(query("SELECT * FROM test_temporal_partitioning WHERE col = " + exampleValue))
+                    .matches("SELECT " + exampleValue);
+        }
+        finally {
+            assertUpdate("DROP MATERIALIZED VIEW test_temporal_partitioning");
+        }
+    }
+
+    @DataProvider
+    public Object[][] testTemporalPartitioningDataProvider()
+    {
+        return new Object[][] {
+                {"year", "date", "DATE '2005-09-09'"},
+                {"year", "timestamp(6)", "TIMESTAMP '2005-09-10 13:31:00.123456'"},
+                {"year", "timestamp(6) with time zone", "TIMESTAMP '2005-09-10 13:00:00.123456 Europe/Warsaw'"},
```

Comment:
For the timestamp with time zone tests, can you add a second value that has the same time but in a different zone and show that works properly?

Reply:
Also, a test with precision > 6?

Reply:
> Also, a test with precision > 6?

this is not supported